src/robot/libraries/Collections.py

```diff
@@ -512,32 +545,13 @@ def _log_list(self, list_):
             for index, item in enumerate(list_):
                 yield f"{index}: {item}"
 
-    def _index_to_int(self, index, empty_to_zero=False):
-        if empty_to_zero and not index:
-            return 0
-        try:
-            return int(index)
-        except ValueError:
-            raise ValueError(f"Cannot convert index '{index}' to an integer.")
-
-    def _index_error(self, list_, index):
+    def _index_error(self, list_: Sequence, index: int) -> NoReturn:
         raise IndexError(f"Given index {index} is out of the range 0-{len(list_) - 1}.")
 
-    def _validate_list(self, list_, position=1):
-        if not is_list_like(list_):
-            raise TypeError(
-                f"Expected argument {position} to be a list or list-like, "
-                f"got {type_name(list_)} instead."
-            )
-
-    def _validate_lists(self, *lists):
-        for index, item in enumerate(lists, start=1):
-            self._validate_list(item, index)
-
 
 class _Dictionary:
 
-    def convert_to_dictionary(self, item):
+    def convert_to_dictionary(self, item: Mapping) -> dict:
```

Comment:
Should use `object` for same reasons as `convert_to_list`. I'm not sure has anyone done so with this keyword, but it's possible to convert also other types than mappings to dicts. For example, `[('k1', 1), ('k2', 2)]` and just `[]` work.

Reply:
The point of this keyword is to convert anything to a `dict`. In that regard it's best if argument conversion doesn't touch the original value at all. Alternatively would be typing all possible convertible values, but as you noted that gets _really_ complicated.

In this kind if cases it's fine to ignore type checker warnings. If you want to add an explicit `type: ignore`, that's fine. Catching and reraising the error with a better error could also be done, but that's not really part of this PR.